sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -357,34 +373,66 @@ case class GpuProjectExec(
    // serde: https://github.com/scala/scala/blob/2.12.x/src/library/scala/collection/
    //   immutable/List.scala#L516
    projectList: List[NamedExpression],
-   child: SparkPlan)(
+   child: SparkPlan,
+   dumpForReplay: Boolean // for testing purpose
+ )(
    useTieredProject : Boolean = false
  ) extends GpuProjectExecLike {
 
+  // for testing purpose; used by dumping for replay feature
+  private val replayDumper: Option[ReplayDumper] = if (dumpForReplay) {
+    val spark = SparkSession.active
+    val hadoopConf = new SerializableConfiguration(spark.sparkContext.hadoopConfiguration)
+    val dumpDir = conf.getConfString(RapidsConf.TEST_REPLAY_EXEC_DUMP_DIR.key)
+    val thresholdMS = conf.getConfString(RapidsConf.TEST_REPLAY_EXEC_THRESHOLD_MS.key).toInt
+    val maxDumpNum = conf.getConfString(RapidsConf.TEST_REPLAY_EXEC_MAX_BATCH_NUM.key).toInt
+    val projectHashCode = this.hashCode()
+    Some(ReplayDumper(hadoopConf, dumpDir, thresholdMS, maxDumpNum, projectHashCode))
+  } else {
+    None
+  }
+
   override def otherCopyArgs: Seq[AnyRef] =
     Seq[AnyRef](useTieredProject.asInstanceOf[java.lang.Boolean])
+
   override def output: Seq[Attribute] = projectList.map(_.toAttribute)
 
   override lazy val additionalMetrics: Map[String, GpuMetric] = Map(
     OP_TIME -> createNanoTimingMetric(MODERATE_LEVEL, DESCRIPTION_OP_TIME))
 
-  override def internalDoExecuteColumnar() : RDD[ColumnarBatch] = {
+  override def internalDoExecuteColumnar(): RDD[ColumnarBatch] = {
```

Comment:
ditto

Reply:
Keep it, because it follows the code format.